src/lib/operators/projection.cpp

```diff
@@ -59,6 +59,27 @@ std::shared_ptr<const Table> Projection::_on_execute() {
   const auto output_table =
       std::make_shared<Table>(column_definitions, output_table_type, input_table_left()->max_chunk_size());
 
+  /**
+   * Identify PQPSelectExpressions that are not correlated and execute them once (instead of once per chunk).
+   */
+
+  auto uncorrelated_select_results = std::make_shared<ExpressionEvaluator::UncorrelatedSelectResults>();
+  {
+    auto evaluator = ExpressionEvaluator{};
+    for (const auto& expression : expressions) {
+      visit_expression(expression, [&](const auto& sub_expression) {
+        const auto pqp_select_expression = std::dynamic_pointer_cast<PQPSelectExpression>(sub_expression);
+        if (pqp_select_expression && pqp_select_expression->parameters.empty()) {
```

Comment:
You have a handful occasions where you call `pqp_select_expression->parameters.empty()` when actually querying "is this pqp_select_expression uncorrelated". 
Maybe add a member "PQPSelectExpression::is_correlated()" for better readability?

Reply:
done, but that raised a question (see below)